src/tests/functional/shared_test_classes/src/subgraph/gather_weights_decompression.cpp

```diff
@@ -0,0 +1,230 @@
+            auto shift_reshape = std::make_shared<ov::op::v1::Reshape>(shift_convert, shift_reshape_const, false);
+            shift_convert = shift_reshape;
+        }
+        mul_parent = std::make_shared<ov::op::v1::Subtract>(weights_convert, shift_convert);
+    }
+
+    ov::test::utils::InputGenerateData in_data;
+    in_data.start_from = -0.5;
+    in_data.range = 1;
+    in_data.resolution = 30000;
+    auto scale_tensor = ov::test::utils::create_and_fill_tensor(output_precision, scaleshift_const_shape, in_data);
+    for (size_t i = 0; i < scale_tensor.get_size(); i++) {
+        if (output_precision == ov::element::f16)
+            scale_tensor.data<ov::float16>()[i] /= ov::float16(16.f);
+        else if (output_precision == ov::element::f32)
+            scale_tensor.data<float>()[i] /= 16.f;
+    }
+    std::shared_ptr<ov::Node> scale_const = std::make_shared<ov::op::v0::Constant>(scale_tensor);
+    if (reshape_on_decompression_constant) {
+        auto scale_reshape_const =
+            ov::op::v0::Constant::create(ov::element::i32, {scaleshift_target_shape.size()}, scaleshift_target_shape);
+        auto scale_reshape = std::make_shared<ov::op::v1::Reshape>(scale_const, scale_reshape_const, false);
+        scale_const = scale_reshape;
+    }
+    std::shared_ptr<ov::Node> last_node = std::make_shared<ov::op::v1::Multiply>(mul_parent, scale_const);
+
+    if (group_decompression) {
+        auto reshape_target_shape = std::vector<int>{static_cast<int>(data_shape[0]), -1};
+        auto target_shape_node =
+            ov::op::v0::Constant::create(ov::element::i32, {reshape_target_shape.size()}, reshape_target_shape);
+        last_node = std::make_shared<ov::op::v1::Reshape>(last_node, target_shape_node, false);
+    }
+    return last_node;
+}
+void GatherWeightsDecompression::generate_inputs(const std::vector<ov::Shape>& target_input_static_shapes) {
+    inputs.clear();
+    const auto& model_inputs = function->inputs();
+    for (size_t i = 0; i < model_inputs.size(); ++i) {
+        const auto& model_input = model_inputs[i];
+        ov::test::utils::InputGenerateData in_data;
+        in_data.start_from = -1;
+        in_data.range = 2;
+        in_data.resolution = 10000;
+        ov::Tensor tensor = ov::test::utils::create_and_fill_tensor(model_input.get_element_type(),
+                                                                    target_input_static_shapes[i],
+                                                                    in_data);
+        inputs.insert({model_input.get_node_shared_ptr(), tensor});
+    }
+}
+void GatherWeightsDecompression::check_results() {
+    const auto& test_param = GetParam();
+    ov::element::Type weights_precision = std::get<2>(test_param);
+    for (const auto& n : compiledModel.get_runtime_model()->get_ordered_ops()) {
+        if (n->get_friendly_name() == "Compressed_weights") {
+            ASSERT_EQ(n->get_output_element_type(0), weights_precision);
+        }
+    }
```

Comment:
What do you think of checking the number of executable nodes? If the pattern match failed, there should be four executable nodes at least, such as convert, multiply, gather and result. But if the pattern match was successful, the convert, multiply and gather nodes were fused into a gathercompressed node. So the number of executable nodes is two. In addition to this, there could be one more node in the case of fp16.

```suggestion
    size_t num_exec_ops = 0;
    for (const auto& n : compiledModel.get_runtime_model()->get_ordered_ops()) {
        if (n->get_input_size() > 0) {
            num_exec_ops += 1;
        }
    }
    EXPECT_TRUE(num_exec_ops <= 3);
```

Reply:
Good idea,
I have test it for CPU, it works fine. For CPU, max `num_exec_ops` is 2. original model min `num_exec_ops `is 4.